tests/test-response-type.js

```diff
@@ -0,0 +1,402 @@
+ * 6) Did a test of xhr.responseType="document" using a simple xml example.
+ */
+async function runTest() {
+  const type = (o) => { return `type=${o?.constructor?.name}`; };
+  try {
+  let r = await upload(xhr, urlF32, F32);  // big
+  log('upload urlF32,    F32      ', r);
+  r = await upload(xhr, urlUtf8, F32Utf8); // big
+  log('upload urlUtf8,   F32Utf8  ', r);
+
+  r = await upload(xhr, urlF32_2, F32_2);
+  log('upload urlF32_2,  F32_2    ', r);
+  r = await upload(xhr, urlUtf8_2, F32Utf8_2);
+  log('upload urlUtf8_2, F32Utf8_2', r);
+
+  r = await upload(xhr, urlXml, JSON.stringify(xmlDoc));
+  log('upload:urlXml,    xmlDoc   ', r);
+
+  const testJson = storageLength();
+  r = await upload(xhr, urlJson, JSON.stringify(testJson));
+  log('upload:urlJson,   storage  ', r);
+
+  log('-----------------------------------------------------------------------------------');
+  checkStorage(); // Check what's in the mini-webserver storage.
+  log('-----------------------------------------------------------------------------------');
+
+  const ab = await download(xhr, urlF32, 'arraybuffer'); // big
+  const f32 = new Float32Array(ab)
+  log('download urlF32    arraybuf', f32.byteLength, type(ab));
+  if (f32.byteLength !== F32.length)
+    throw new Error(`Download from urlF32 has incorrect length: ${f32.byteLength} !== ${F32.length}`);
+
+  const text1 = await download(xhr, urlUtf8, 'text'); // big
+  log('download urlUtf8       text', text1.length, type(text1));
+  if (f32.byteLength !== text1.length)
+    throw new Error(`Download from urlUtf8 has incorrect length: ${f32.byteLength} !== ${text1.length}`);
+
+  if (typeof Blob === 'function') { // Check to see if Blob exists.
+    const blob = await download(xhr, urlF32_2, 'blob');
+    const blob_ab = await blob.arrayBuffer();
+    const blob_f32 = new Float32Array(blob_ab);
+    log(`download urlF32_2      blob ${blob_f32.byteLength} ${type(blob)}  `, blob_f32);
+    if (!isEqual(blob_f32, _f32_2))
+      throw new Error(`Download from urlF32_2 has incorrect content: ${blob_f32} !== ${_f32_2}`);
+  } else {
+    // If Blob doesn't exist (node versions < 18), then use xhr.responseType="arraybuffer".
+    const ab_2 = await download(xhr, urlF32_2, 'arraybuffer');
+    const ab_2_f32 = new Float32Array(ab_2);
+    log(`download urlF32_2  arraybuf ${ab_2_f32.byteLength} ${type(ab_2)}`, ab_2_f32);
+    if (!isEqual(ab_2_f32, _f32_2))
+      throw new Error(`Download from urlF32_2 has incorrect content: ${ab_2_f32} !== ${_f32_2}`);
+  }
+
+  const text2 = await download(xhr, urlUtf8_2, '');
+  const text2_f32 = stringToFloat32Array(text2);
+  log('download urlUtf8_2  default', text2.length, type(text2), text2_f32);
+  if (!isEqual(text2_f32, _f32_2))
+    throw new Error(`Download from urlUtf8_2 has incorrect content: ${text2_f32} !== ${_f32_2}`);
+
+  try {
+    const { XMLSerializer } = require('@xmldom/xmldom'); // Throws when xmldom not installed.
```

Comment:
Do you think this warrants the addition of an optional `peerDependency` on `@xmldom/xmldom`?